e2e_playwright/multipage_apps_v2/mpa_v2_basics_test.py

```diff
@@ -0,0 +1,297 @@
+    expect(nav.locator("a").nth(0)).to_contain_text("page 2")
+    expect(nav.locator("a").nth(1)).to_contain_text("Different Title")
+    expect(nav.locator("a").nth(2)).to_contain_text("page 4")
+    expect(nav.locator("a").nth(3)).to_contain_text("page 5")
+    expect(nav.locator("a").nth(4)).to_contain_text("page 6")
+    expect(nav.locator("a").nth(5)).to_contain_text("page 7")
+    expect(nav.locator("a").nth(6)).to_contain_text("page 8")
+    expect(nav.locator("a").nth(7)).to_contain_text("page 9")
+    expect(nav.locator("a").nth(8)).to_contain_text("page 10")
+    expect(nav.locator("a").nth(9)).to_contain_text("page 11")
+    expect(nav.locator("a").nth(10)).to_contain_text("page 12")
+
+
+def test_dynamic_pages(themed_app: Page, assert_snapshot: ImageCompareFunction):
+    """Test that dynamic pages are defined"""
+    themed_app.get_by_test_id("stCheckbox").nth(1).click(delay=50)
+    wait_for_app_run(themed_app)
+
+    nav = themed_app.get_by_test_id("stSidebarNav")
+    expect(nav.locator("a").nth(0)).to_contain_text("page 2")
+    expect(nav.locator("a").nth(1)).to_contain_text("Different Title")
+    expect(nav.locator("a").nth(2)).to_contain_text("page 5")
+    expect(nav.locator("a").nth(3)).to_contain_text("page 9")
+
+    assert_snapshot(nav, name="dynamic-pages")
+
+
+def test_show_not_found_dialog(page: Page, app_port: int):
+    """Test that we show a not found dialog if the page doesn't exist."""
+    page.goto(f"http://localhost:{app_port}/not_a_page")
+    wait_for_app_loaded(page)
+
+    expect(page.locator('[role="dialog"]')).to_contain_text("Page not found")
+
+
+def test_handles_expand_collapse_of_mpa_nav_correctly(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test that we handle expand/collapse of MPA nav correctly."""
+
+    themed_app.get_by_test_id("stCheckbox").nth(2).click(delay=50)
+    wait_for_app_run(themed_app)
+
+    view_button = themed_app.get_by_test_id("stSidebarNavViewButton")
+
+    expect(view_button).to_be_visible()
+
+    # Expand the nav
+    view_button.click(force=True)
+    # We apply a quick timeout here so that the UI has some time to
+    # adjust for the screenshot after the click
+    themed_app.wait_for_timeout(250)
+    assert_snapshot(
+        themed_app.get_by_test_id("stSidebarNav"), name="mpa-sidebar_nav_expanded"
+    )
+
+    # Collapse the nav
+    view_button.click(force=True)
+    themed_app.wait_for_timeout(250)
+    assert_snapshot(
+        themed_app.get_by_test_id("stSidebarNav"), name="mpa-sidebar_nav_collapsed"
+    )
+
+    # Expand the nav again
+    view_button.click(force=True)
+    themed_app.wait_for_timeout(250)
+    assert_snapshot(
+        themed_app.get_by_test_id("stSidebarNav"), name="mpa-sidebar_nav_expanded"
+    )
+
+
+def test_switch_page_by_path(app: Page):
+    """Test that we can switch between pages by triggering st.switch_page with a path."""
+
+    app.get_by_test_id("stButton").nth(0).locator("button").first.click()
```

Comment:
I would suggest to use a filter at places like this to get the correct button, `filter(has_text="page 5")`. This makes it much easier to read the test and understand what exactly is supposed to happen